proto/p4/config/p4info.proto

```diff
@@ -21,13 +21,11 @@ import "google/protobuf/any.proto";
 package p4.config;
 
 message P4Info {
-  repeated HeaderField header_fields = 1;
```

Comment:
In order to make this work without header fields in the P4Info, we will need to store the HF IDs in our proto blob for the switch.  Where in the p4c IR does the authoritative ID exist, i.e. the one that would appear in the P4Info header field if one was present?  Is this ID present before the serializer runs, or does it appear after.  Does the serializer make ID decisions such that we could get different ID output depending on whether the command-line chooses to invoke the serializer?